Use react-inspector for reps in most cases

mdboom opened the pull request:
See #781 for screenshots and justification.

This changes most of the basic reps and the object reps to use react-inspector.  They are generally more vertically-space-efficient and self-consistent than what we had before.

The array, matrix, data frame, error, promise and custom reps are retained from before since react-inspector doesn't handle those.  In separate PRs, these will get updated to be more stylistically consistent with what react-inspector does, but for now I think we merge this because it doesn't really regress anything.

Of special note is that react-inspector's object inspector is able to display iodide custom reps as values.  Conversely, our collection reps use react-inspector's reps when appropriate.

mdboom commented:
I think the test failures are server related...

bcolloran commented on src/components/reps/value-renderer.jsx:
```diff
@@ -122,39 +111,34 @@ const complexHandlers = [
   // insert new handlers *after* the scalar handlers
-  userHandlers.unshift(wrapHandler(handler))
-  handlers = simpleTypeHandlers.concat(userHandlers, complexHandlers)
+  userHandlers.unshift(handler)
```
why'd you remove the wrapHandler here? don't we especially need to wrap user-added handlers in try/catch?

mdboom replied:
Sorry -- this was just left in after experimenting with something...

bcolloran commented on src/components/reps/value-renderer.jsx:
```diff
@@ -32,10 +25,13 @@ export function renderValue(value, inContainer = false) {
         return resultElem
       }
-      console.warn(`Unknown output handler result type from ${handler}`)
+      console.warn(`Unknown out handler result type from ${handler}`)
```
"out handler"?

mdboom replied:
Just an oops.  Reverted.

bcolloran commented:
big improvement over json tree! couple other thoughts

- as mentioned in #781, the long string rep does work for me, so perhaps we should not cut that yet?
- when i was hacking on reps last month, i found this array snippet useful as a test case:
```
// arrays with mixed types
let array = []
for (let i = 0; i < 100; ++i) {
  array.push(i)
  array.push('_"'+i)
  array.push(undefined)
  array.push(false)
  array.push(NaN)
  array.push(null) 
  array.push(i*Math.random())
  array.push(() => i)
}
array
```

- even tho the tests pass, i see this when i run them:
```
 PASS  src/components/reps/__tests__/promise-handler.test.jsx
  ● Console

    console.error src/components/reps/value-renderer.jsx:84
      output handler error TypeError: Cannot read property 'shouldHandle' of undefined
          at Object.shouldHandle (/home/bcolloran/iodide/iodide/src/components/reps/value-renderer.jsx:139:87)
          at renderValue (/home/bcolloran/iodide/iodide/src/components/reps/value-renderer.jsx:123:1082)
          at PromiseRep.render (/home/bcolloran/iodide/iodide/src/components/reps/promise-handler.jsx:40:129)
          at ReactShallowRenderer._updateClassComponent (/home/bcolloran/iodide/iodide/node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:238:39)
          at ReactShallowRenderer.render (/home/bcolloran/iodide/iodide/node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:110:12)
          at Updater.enqueueSetState (/home/bcolloran/iodide/iodide/node_modules/enzyme-adapter-react-16/node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:321:20)
          at PromiseRep.Object.<anonymous>.Component.setState (/home/bcolloran/iodide/iodide/node_modules/react/cjs/react.development.js:238:16)
          at PromiseRep.state.promise.props.promise.then.val (/home/bcolloran/iodide/iodide/src/components/reps/promise-handler.jsx:16:12)
          at <anonymous>
          at process._tickCallback (internal/process/next_tick.js:188:7)
```

mdboom commented:
I was able to get the tests to work by forcing `ValueRenderer` to import before the specific output handlers being tested.  Not sure if there's a better solution.

mdboom commented:
I've updated this to reinstate the old string rep, but adjusted the styles (and how it gets used) to blend in with react-inspector.

I also discovered some CSS styles that are obsoleted by this PR and I removed those.

hamilton commented:
Hmm - it seems something else is at play here now. I'm finding that long strings still break the console, so it might actually not be the rep itself but the console not having `overflow-x` in the return value DOM element set properly. I know it used to be before the iframe migration.

This causes an overflow in the console, as far as I can tell.

```javascript
var randomString = (length=100) => {
  var text = "";
  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

  for (var i = 0; i < length; i++)
    text += possible.charAt(Math.floor(Math.random() * possible.length));

  return text;
}

let a = randomString(1000)
a
```

mdboom commented:
I see that, too.  I thought I had fixed that exact issue -- I'm looking into it.

hamilton commented:
@mdboom this is great! The string overflow issue has been fixed w/ the latest commit, and the string-wrapping + truncation is MUCH more useful for debugging large strings (which users will inevitably encounter when they `fetch` a csv without parsing it).

This appears ready to merge to me.
